source/retryable-writes/retryable-writes.rst

```diff
@@ -516,8 +516,24 @@ The above rules are implemented in the following pseudo-code:
         throw previousError;
       }
 
-      /* CSOT is enabled and the operation has timed out. */
-      if (timeoutMS != null && isExpired(timeoutMS) {
+      if (timeoutMS == null) {
+        /* If CSOT is not enabled, allow any retryable error from the second
+         * attempt to propagate to our caller, as it will be just as relevant
+         * (if not more relevant) than the original error. For exceptions that
+         * originate from the driver, we should raise the previous error. Other
+         * exceptions originating from the server should be allowed to
+         * propagate. */
+        try {
+          return executeCommand(server, retryableCommand);
+        } catch (DriverException ignoredError) {
+          throw previousError;
+        } catch (Exception secondError) {
+          handleError(secondError);
+          throw secondError;
+        }
+        break
```

Comment:
This doesn't account for NoWritesPerformed or error code 20. Is it possible to avoid duplicating the call to executeCommand and other logic here? Perhaps:
```
if ((timeoutMS == null && retrying) || (timeoutMS != null && isExpired(timeoutMS))) {
    throw previousError;
}
retrying = True;
```